cyclic_boosting/quantile_matching.py

```diff
@@ -161,6 +166,571 @@ def cdf(self, x):
         )
 
 
+class sinhlogistic:
+    """
+    sinh/arcsinh-modified logistic distribution for smooth interpolation
+    between logistic and t2 distribution.
+
+    Parameters
+    ----------
+    shape: float
+        parameter modifying the logistic base distribution via
+        sinh/arcsinh-scaling
+    """
+
+    def __init__(self, shape: float):
+        self.shape = shape
+
+    def ppf(self, x):
+        # ppf of natural logistic distribution
+        xlog = 0.25 * np.log(x / (1.0 - x))
+
+        # sinh or arcsinh scaling
+        if self.shape > 0:
+            x = np.arcsinh(self.shape * xlog) / self.shape
+        elif self.shape < 0:
+            x = np.sinh(self.shape * xlog) / self.shape
+        else:
+            x = xlog
+        return x
+
+    def cdf(self, x):
+        # sinh or arcsinh scaling
+        if self.shape > 0:
+            xlog = np.sinh(self.shape * x) / self.shape
+        elif self.shape < 0:
+            xlog = np.arcsinh(self.shape * x) / self.shape
+        else:
+            xlog = x
+
+        # natural logistic cdf
+        return 1.0 / (1 + np.exp(-4 * xlog))
+
+    def pdf(self, x):
+        if self.shape > 0:
+            return np.cosh(self.shape * x) / (np.cosh(2.0 / self.shape * np.sinh(self.shape * x))) ** 2
+        elif self.shape < 0:
+            return (
+                1.0 / np.sqrt((self.shape * x) ** 2 + 1) / (np.cosh(2.0 / self.shape * np.arcsinh(self.shape * x))) ** 2
+            )
+        else:
+            return 1.0 / (np.cosh(2 * x)) ** 2
+
+
+class basedist:
+    """
+    Scaling of base ppfs such that ppf(1 - alpha) = 1. A detailed description
+    can be found in the presentation JQPDregression.pdf in the docs folder of
+    this repository.
+    """
+
+    def __init__(self, dist, alpha):
```

Comment:
Missing types.

Reply:
From what I can tell, the dist type comes from scipy and it is not very well defined there, so I guess we can omit it here as well